test/functional/feature_assumeutxo.py

```diff
@@ -218,6 +222,27 @@ def test_snapshot_block_invalidated(self, dump_output_path):
             assert_raises_rpc_error(-32603, msg, node.loadtxoutset, dump_output_path)
             node.reconsiderblock(block_hash)
 
+    def test_snapshot_not_on_most_work_chain(self, dump_output_path):
+        self.log.info("Test bitcoind should fail when the node knows the headers of another chain with more work.")
```

Comment:
I think this test also actually resolves the TODOs that https://github.com/bitcoin/bitcoin/pull/29996/commits/556ff68cea1dfa6c444ef9c920dd0ce52ecc9668 aimed to resolve. I will elaborate more there.

Reply:
Can remove the TODO `Valid snapshot file and snapshot block, but the block is not on the most-work chain` here now as well.

Reply:
Actually also this TODO, I think these basically mean the same thing: `- TODO: Not an ancestor or a descendant of the snapshot block and has more work`

Reply:
Could also mention #28648 as related in description because of this.

Reply:
Removed both TODOs.